File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-3.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        def dfs(root):
            if not root:
                return []
            
            left, right = dfs(root.left), dfs(root.right)          
            output = [root]

            for node in left:
                if node.val >= root.val:
                    output.append(node)
            
            for node in right:
                if node.val >= root.val:
                    output.append(node)

            return output
        
        return len(dfs(root))

        

File: Data Structures & Algorithms/leetcode/two-pointers/two-integer-sum-ii/submission-3.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int i = 0;
        int j = numbers.length-1;
        int[] output = new int[2]; 

        while (i < j && numbers[i] + numbers[j] != target) {
            if (numbers[i] + numbers[j] < target) {
                i++;
            }
            else {
                j--;
            }
        }

        output[0] = i+1;
        output[1] = j+1;

        return output;
    }
}
